### ISSUE TITLE
Function DocumentReference.update() called with invalid data. Unsupported field value: a custom File object

### ISSUE THREAD

**Issue by mzgnr** [benwinding/react-admin-firebase#38]
Function DocumentReference.update() called with invalid data. Unsupported field value: a custom File object

**Comment by mzgnr** [comment]
Fileupload not working

**Comment by benwinding** [comment]
Can you please provide more information? 

What does the resource structure look like that you're working with? Could you share some code?

**Comment by mzgnr** [comment]
```jsx
<ArrayInput source="tests">
          <SimpleFormIterator>
            <DateInput
              options={{ format: "DD/MM/YYYY" }}
              source="date"
              label="Date"
            />
            <ReferenceInput
              label="Select Test Type"
              source="test_id"
              reference="tests"
            >
              <SelectInput optionText="name" />
            </ReferenceInput>
            <FileInput
              source="file"
              label="Documents"
              multiple
              accept="application/pdf, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel"
            >
              <FileField source="src" title="title" />
            </FileInput>
            <LongTextInput source="notes" label="Notes" />
          </SimpleFormIterator>
        </ArrayInput>
```

**Comment by benwinding** [comment]
Ah, you discovered a limitation. Currently this package doesn't support `<FileField />` within the `<ArrayInput/>` fields. 

It should definitely be supported, so I'll try and implement this in the next few days.

**Comment by mzgnr** [comment]
@benwinding i can help you if you tell me which lines to look at, i can implement it

**Comment by mzgnr** [comment]
@benwinding btw, multiple file option is also not working

**Comment by benwinding** [comment]
Hi @mzgnr,

The original issue should be fixed in `v0.8.6`, so `<ArrayInput />` works with `<FileInput />` (one level of nesting at this stage).

I wasn't aware there was a _multiple file option_ on the FileField and will need to check this out later.

Cheers,
Ben

**Comment by mzgnr** [comment]
Thanks for the update. I switched to my own client library and implemented these features.
